src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -596,12 +612,23 @@ def compute_nhd_routing_v02(
     else:  # Execute in serial
         results = []
         for twi, (tw, reach_list) in enumerate(reaches_bytw.items(), 1):
-            segs = list(chain.from_iterable(reach_list))
+            segs = sorted(chain.from_iterable(reach_list))
```

Comment:
@groutr @jhrehanoaa 
Sorting here reduces the number of index sorts needed later --- but then the segs loses the order-of-execution, which could be useful. See the comment @jdmattern-noaa just below...

Reply:
@jameshalgren you can generate the order-of-execution from the reach-list at any time. In this particular context, it isn't used, so I don't see why there is worry about losing it.

Reply:
Good points, @groutr. The (possible) need became apparent in this soon-to-be-incoming PR: 
https://github.com/jdmattern-noaa/t-route/blob/935e130c9ed1f78675fc76d42d4e9722d18af543/src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py#L616
(Thanks, @jdmattern-noaa for putting this together.)

The idea there is that by using segs in execution order to filter the waterbodies, we can use a simple index inside the compute network loop to walk through the waterbody calculations as they are encountered in the sequence of reach calculations.

I like your suggestion of using another variable -- that's simple enough.